content/rest-api/index.textile

```diff
@@ -461,6 +469,251 @@ bc[json]. [{
+    metadata: <object>,
+    state: <active or failed>
+  }
+}
+
+h3(#get-device-registrations). List registered devices
+
+h6. GET rest.ably.io/push/deviceRegistrations
+
+Obtain the details for devices registered for receiving push registrations.
+
+Example request:
+
+bc[sh]. curl https://rest.ably.io/push/deviceRegistrations \
+ -u "{{API_KEY}}"
+
+h5. Parameters
+
+- clientId := optional filter to restrict to devices associated with that @clientId@
+- limit := _100_ The maximum number of records to return. A limit greater than 1,000 is invalid.
+
+h5. Options
+
+- Content-Type := not applicable
+- Accept := @application/json@ by default, or @application/x-msgpack@, @text/html@
+- Auth required := yes ("basic":#basic-authentication or "token":#token-authentication)
+
+h5. Returns
+
+A successful request returns a "paginated response":#pagination of:
+
+bc[json]. [{
+  id: <string>,
+  clientId: <optional, string>
+  platform: <string>
+  formFactor: <string>,
+  metadata: <object>,
+  updateToken: <string>,
+  push: {
+    transportType: <string>,
+    metadata: <object>,
+    state: <active or failed>
+  }
+}]
+
+h3(#get-channel-subscriptions). List channel subscriptions
+
+h6. GET rest.ably.io/push/channelSubscriptions
+
+Obtain push notification subscriptions to channels.
+
+Example request:
+
+bc[sh]. curl https://rest.ably.io/push/channelSubscriptions \
+ -u "{{API_KEY}}"
+
+h5. Parameters
+
+- clientId := Filter to restrict to subscriptions associated with that @clientId@. Cannot be set when @deviceId@ is set.
+- deviceId := Filter to restrict to subscriptions for that @devideId@. Cannot be set when @clientId@ is set.
+- limit := _100_ The maximum number of records to return. A limit greater than 1,000 is invalid.
+
+h5. Options
+
+- Content-Type := not applicable
+- Accept := @application/json@ by default, or @application/x-msgpack@, @text/html@
+- Auth required := yes ("basic":#basic-authentication or "token":#token-authentication)
+
+h5. Returns
+
+A successful request returns a "paginated response":#pagination of:
+
+bc[json]. [{
+  deviceId: <string, only present when clientId is not present>
+  clientId: <string, only present when deviceId is not present>
+  channel: <string>
+}]
+
+h3(#delete-device-registration). Unregister device for push notifications
+
+h6. DELETE rest.ably.io/push/deviceRegistrations/@<device id>@
+
+Unregister device. All its subscriptions will be removed.
+
+Example request:
+
+bc[sh]. curl -X DELETE https://rest.ably.io/push/deviceRegistrations/01ARZ3NDEKTSV4RRFFQ69G5FAV \
+ -u "{{API_KEY}}"
+
+h5. Parameters
+
+None.
+
+h5. Options
+
+- Content-Type := not applicable
+- Auth required := yes ("basic":#basic-authentication or "token":#token-authentication)
+
+h5. Returns
+
+A successful request returns nothing.
+
+An unsuccessful request returns an error.
+
+h3(#delete-channel-subscription). Unsubscribe from push notifications from channel
+
+h6. DELETE rest.ably.io/push/channelSubscriptions
```

Comment:
> Delete all subscriptiosn for this clientId

`DELETE /push/channelSubscriptions?deviceId=...`

> Delete all devices and associated subscriptions matching this deviceId

I guess here "all devices" is only one device, given deviceId is unique? Then `DELETE /push/deviceRegistrations/<deviceId>`.

> Delete all subscriptions for this device
> Delete all devices and associated subscriptiosn matching this clientId

Those are not implemented but should be straightforward. Added to the backlog.